src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,178 @@
+package duke.ui;
+
+import java.util.Scanner;
+
+import duke.tasklist.TaskList;
+import duke.tasktypes.Task;
+
+/**
+ * Represents a User Interface (Ui) instance.
+ */
+public class Ui {
+
+    /** Scanner Object for reading User input. */
+    private Scanner sc;
+
+    public Ui() {
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * Closes the Scanner object
+     */
+    public void close() {
+        sc.close();
+    }
```

Comment:
I try to use the UI instance as much as possible, instead of interacting with the Scanner object directly. Hence the wrapping of ```sc.close()``` in a ```ui.close()``` method.